hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java

```diff
@@ -0,0 +1,167 @@
+     * {@link com.hazelcast.jet.core.processor.SinkProcessors#writeJmsTopicP}
+     * instead
+     */
+    public static ProcessorMetaSupplier supplier(DistributedSupplier<ConnectionFactory> factorySupplier,
+                                                 String name, boolean isTopic) {
+        return supplier(
+                () -> uncheckCall(() -> factorySupplier.get().createConnection()),
+                connection -> uncheckCall(() -> connection.createSession(false, Session.AUTO_ACKNOWLEDGE)),
+                (session, item) -> uncheckCall(() -> session.createTextMessage(item.toString())),
+                (producer, message) -> uncheckRun(() -> producer.send(message)),
+                noopConsumer(), name, isTopic
+        );
+    }
+
+    private static final class Supplier<T> implements ProcessorSupplier {
+
+        static final long serialVersionUID = 1L;
+
+        private final DistributedSupplier<Connection> connectionSupplier;
+        private final DistributedFunction<Connection, Session> sessionF;
+        private final String name;
+        private final boolean isTopic;
+        private final DistributedBiFunction<Session, T, Message> messageFn;
+        private final DistributedBiConsumer<MessageProducer, Message> sendFn;
+        private final DistributedConsumer<Session> flushFn;
+
+        private transient Connection connection;
+
+        private Supplier(DistributedSupplier<Connection> connectionSupplier,
+                         DistributedFunction<Connection, Session> sessionF,
+                         DistributedBiFunction<Session, T, Message> messageFn,
+                         DistributedBiConsumer<MessageProducer, Message> sendFn,
+                         DistributedConsumer<Session> flushFn,
+                         String name,
+                         boolean isTopic
+        ) {
+            this.connectionSupplier = connectionSupplier;
+            this.sessionF = sessionF;
+            this.messageFn = messageFn;
+            this.sendFn = sendFn;
+            this.flushFn = flushFn;
+            this.name = name;
+            this.isTopic = isTopic;
+        }
+
+        @Override
+        public void init(Context ignored) {
+            connection = connectionSupplier.get();
+            uncheckRun(() -> connection.start());
+        }
+
+        @Override
+        public Collection<? extends Processor> get(int count) {
+            DistributedFunction<Processor.Context, JmsContext> createFn = jet -> {
+                Session session = sessionF.apply(connection);
+                Destination destination =
+                        uncheckCall(() -> isTopic ? session.createTopic(name) : session.createQueue(name));
+                MessageProducer producer = uncheckCall(() -> session.createProducer(destination));
+                return new JmsContext(session, producer);
+            };
+            DistributedBiConsumer<JmsContext, T> onReceiveFn = (buffer, item) -> {
```

Comment:
`buffer` variable name is misleading here (also occurs below), should be `jmsContext`.

Reply:
You're right, but this PR is merged. Feel free to submit a PR :)